File: CS/GridControlDetailView/CustomGrid/CustomViewInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraGrid.Views.Base.ViewInfo;
using DevExpress.XtraGrid.Views.Base;

namespace CustomGrid
{
    public class CustomViewInfo : BaseViewInfo
    {
        public CustomViewInfo(CustomView view)
            : base(view)
        {
            IsReady = true;
        }

        public override BaseView View
        {
            get
            {
                return base.View;
            }
        }

        protected override void AddAnimatedItems()
        {
            throw new NotImplementedException();
        }

        public override System.Drawing.Rectangle Bounds
        {
            get
            {
                return View.ViewRect;
            }
        }

        public override System.Drawing.Rectangle ClientBounds
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        protected override DevExpress.Utils.BaseAppearanceCollection CreatePaintAppearances()
        {
            return new DevExpress.Utils.BaseAppearanceCollection();
        }

        protected override BaseSelectionInfo CreateSelectionInfo()
        {
            return new CustomSelectionInfo(View as CustomView);
        }

        protected override DevExpress.XtraEditors.ViewInfo.BaseEditViewInfo HasItem(CellId id)
        {
            throw new NotImplementedException();
        }

        internal void UpdateTabControlInternal() {
            UpdateTabControl();
        }

        protected override void UpdateTabControl() {
            base.UpdateTabControl();
            TabControl.Bounds = CalcBorderRect(Bounds);
        }
    }

    internal class NullTreeListViewInfo : CustomViewInfo
    {
        public NullTreeListViewInfo(CustomView treeListView)
            : base(treeListView)
        {
        }

        public override bool IsNull
        {
            get
            {
                return true;
            }
        }
    }
}

File: VB/GridControlDetailView/CustomGrid/CustomViewInfo.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraGrid.Views.Base.ViewInfo
Imports DevExpress.XtraGrid.Views.Base

Namespace CustomGrid
	Public Class CustomViewInfo
		Inherits BaseViewInfo

		Public Sub New(ByVal view As CustomView)
			MyBase.New(view)
			IsReady = True
		End Sub

		Public Overrides ReadOnly Property View() As BaseView
			Get
				Return MyBase.View
			End Get
		End Property

		Protected Overrides Sub AddAnimatedItems()
			Throw New NotImplementedException()
		End Sub

		Public Overrides ReadOnly Property Bounds() As System.Drawing.Rectangle
			Get
				Return View.ViewRect
			End Get
		End Property

		Public Overrides ReadOnly Property ClientBounds() As System.Drawing.Rectangle
			Get
				Throw New NotImplementedException()
			End Get
		End Property

		Protected Overrides Function CreatePaintAppearances() As DevExpress.Utils.BaseAppearanceCollection
			Return New DevExpress.Utils.BaseAppearanceCollection()
		End Function

		Protected Overrides Function CreateSelectionInfo() As BaseSelectionInfo
			Return New CustomSelectionInfo(TryCast(View, CustomView))
		End Function

		Protected Overrides Function HasItem(ByVal id As CellId) As DevExpress.XtraEditors.ViewInfo.BaseEditViewInfo
			Throw New NotImplementedException()
		End Function

		Friend Sub UpdateTabControlInternal()
			UpdateTabControl()
		End Sub

		Protected Overrides Sub UpdateTabControl()
			MyBase.UpdateTabControl()
			TabControl.Bounds = CalcBorderRect(Bounds)
		End Sub
	End Class

	Friend Class NullTreeListViewInfo
		Inherits CustomViewInfo

		Public Sub New(ByVal treeListView As CustomView)
			MyBase.New(treeListView)
		End Sub

		Public Overrides ReadOnly Property IsNull() As Boolean
			Get
				Return True
			End Get
		End Property
	End Class
End Namespace
